File: shared/src/main/java/com/runtracker/shared/data/model/SafetySettings.kt
package com.runtracker.shared.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.TypeConverters
import com.runtracker.shared.data.db.Converters

@Entity(tableName = "safety_settings")
@TypeConverters(Converters::class)
data class SafetySettings(
    @PrimaryKey
    val id: Long = 1, // Singleton - only one settings record
    val isEnabled: Boolean = true,
    val emergencyContacts: List<EmergencyContact> = emptyList(),
    val sosMessage: String = "I don't feel safe. This is my current location:",
    val checkInEnabled: Boolean = false,
    val defaultCheckInMinutes: Int = 60,
    val panicAlarmEnabled: Boolean = true,
    val fakeCallEnabled: Boolean = true,
    val fakeCallerName: String = "Mom",
    val fakeCallDelay: Int = 5, // seconds delay before fake call
    val autoShareLocationOnSos: Boolean = true,
    val sosCountdownSeconds: Int = 5, // countdown before SOS is sent (can cancel)
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis()
)

data class EmergencyContact(
    val id: String = java.util.UUID.randomUUID().toString(),
    val name: String,
    val phoneNumber: String,
    val relationship: String = "", // e.g., "Partner", "Parent", "Friend"
    val isPrimary: Boolean = false, // Primary contact gets called, others get SMS
    val notifyOnSos: Boolean = true,
    val notifyOnCheckInMissed: Boolean = true
)

@Entity(tableName = "check_in_sessions")
data class CheckInSession(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val startTime: Long,
    val expectedDurationMinutes: Int,
    val expectedEndTime: Long,
    val activityType: String, // RUNNING, SWIMMING, CYCLING
    val isActive: Boolean = true,
    val checkedIn: Boolean = false,
    val checkedInTime: Long? = null,
    val sosTriggered: Boolean = false,
    val sosTriggeredTime: Long? = null,
    val lastKnownLatitude: Double? = null,
    val lastKnownLongitude: Double? = null
)

enum class SafetyAlertType {
    SOS,
    PANIC_ALARM,
    CHECK_IN_MISSED,
    FAKE_CALL
}

File: iosApp/GoSteady/Sources/Models/SafetySettings.swift
import Foundation
import SwiftData

@Model
final class SafetySettings {
    @Attribute(.unique) var id: Int
    var isEnabled: Bool
    var emergencyContacts: [EmergencyContact]
    var sosMessage: String
    var checkInEnabled: Bool
    var defaultCheckInMinutes: Int
    var panicAlarmEnabled: Bool
    var fakeCallEnabled: Bool
    var fakeCallerName: String
    var fakeCallDelay: Int
    var autoShareLocationOnSos: Bool
    var sosCountdownSeconds: Int
    var createdAt: Date
    var updatedAt: Date

    init(
        id: Int = 1,
        isEnabled: Bool = true,
        emergencyContacts: [EmergencyContact] = [],
        sosMessage: String = "I don't feel safe. This is my current location:",
        checkInEnabled: Bool = false,
        defaultCheckInMinutes: Int = 60,
        panicAlarmEnabled: Bool = true,
        fakeCallEnabled: Bool = true,
        fakeCallerName: String = "Mom",
        fakeCallDelay: Int = 5,
        autoShareLocationOnSos: Bool = true,
        sosCountdownSeconds: Int = 5,
        createdAt: Date = Date(),
        updatedAt: Date = Date()
    ) {
        self.id = id
        self.isEnabled = isEnabled
        self.emergencyContacts = emergencyContacts
        self.sosMessage = sosMessage
        self.checkInEnabled = checkInEnabled
        self.defaultCheckInMinutes = defaultCheckInMinutes
        self.panicAlarmEnabled = panicAlarmEnabled
        self.fakeCallEnabled = fakeCallEnabled
        self.fakeCallerName = fakeCallerName
        self.fakeCallDelay = fakeCallDelay
        self.autoShareLocationOnSos = autoShareLocationOnSos
        self.sosCountdownSeconds = sosCountdownSeconds
        self.createdAt = createdAt
        self.updatedAt = updatedAt
    }
}

struct EmergencyContact: Codable, Hashable, Identifiable {
    var id: String
    var name: String
    var phoneNumber: String
    var relationship: String
    var isPrimary: Bool
    var notifyOnSos: Bool
    var notifyOnCheckInMissed: Bool

    init(
        id: String = UUID().uuidString,
        name: String,
        phoneNumber: String,
        relationship: String = "",
        isPrimary: Bool = false,
        notifyOnSos: Bool = true,
        notifyOnCheckInMissed: Bool = true
    ) {
        self.id = id
        self.name = name
        self.phoneNumber = phoneNumber
        self.relationship = relationship
        self.isPrimary = isPrimary
        self.notifyOnSos = notifyOnSos
        self.notifyOnCheckInMissed = notifyOnCheckInMissed
    }
}

enum SafetyAlertType: String, Codable, CaseIterable {
    case sos = "SOS"
    case panicAlarm = "PANIC_ALARM"
    case checkInMissed = "CHECK_IN_MISSED"
    case fakeCall = "FAKE_CALL"
}
